lib/services/breezlib/breez_bridge.dart

```diff
@@ -778,8 +779,11 @@ class BreezBridge {
     if (address == null) {
       return Future.error("empty address");
     }
-    if (addr.toLowerCase().startsWith("bitcoin:")) {
+    addr = extractBitcoinAddressFromBip21(address) ?? addr;
```

Comment:
Looks like we can reduce this whole logic into one function called `extractBitcoinFromAddress(String address)`. I can't really tell the difference between these two as `split` handles the case where `?` is non-existent. 

```
static const String URN_SCHEME = "bitcoin:";

String extractBitcoinAddress(String address) {
    if (address.toLowerCase().startsWith(URN_SCHEME)) {
      try {
        int split = address.indexOf("?");
        String addr =
            address.substring(URN_SCHEME.length, split == -1 ? null : split);
        if (addr.isNotEmpty) {
          isLegacyOrNestedSegwit(addr) ? addr : addr = addr.toLowerCase();
          return addr;
        }
      } on FormatException {
        // do nothing.
      }
    }
    return address;
  }
```
and use it in validateAddress as:

```
Future<String> validateAddress(String address) {
    if (address == null) {
      return Future.error("empty address");
    }
    return _invokeMethodWhenReady("validateAddress", {
      "argument": extractBitcoinAddress(address),
    }).then((address) => address as String);
  }
```

Reply:
Thanks good idea! Fixed